aioredis/client.py

```diff
@@ -1054,14 +1054,23 @@ async def __aenter__(self: _RedisT) -> _RedisT:
     async def __aexit__(self, exc_type, exc_value, traceback):
         await self.close()
 
-    def __del__(self):
+    _DEL_MESSAGE = "Unclosed Redis client"
+
+    def __del__(self, _warnings: Any = warnings) -> None:
         try:
-            loop = asyncio.get_event_loop()
-            if loop.is_running():
-                loop.create_task(self.close())
-            else:
-                loop.run_until_complete(self.close())
-        except Exception:
+            if self.connection is not None:
+                _warnings.warn(
+                    f"Unclosed client session {self!r}",
+                    ResourceWarning,
+                    source=self,
+                )
+                context = {"client": self, "message": self._DEL_MESSAGE}
+                if self._source_traceback is not None:
+                    context["source_traceback"] = self._source_traceback
+                self._loop.call_exception_handler(context)
```

Comment:
Not sure what `_source_traceback` is, so maybe can just drop those 2 lines.
Then `self._loop` will probably need replacing with `asyncio.get_running_loop()`.